include/library/spdm_requester_lib.h

```diff
@@ -346,6 +346,26 @@ libspdm_return_t libspdm_get_measurement_ex2(void *spdm_context, const uint32_t
                                              void *responder_nonce,
                                              void *opaque_data,
                                              size_t *opaque_data_size);
+
+/**
+ * This function sends GET_MEASUREMENT_EXTENSION_LOG to get MEL from device.
+ *
+ * @param  spdm_context               A pointer to the SPDM context.
+ * @param  session_id                 Indicates if it is a secured message protected via SPDM session.
+ *                                    If session_id is NULL, it is a normal message.
+ *                                    If session_id is not NULL, it is a secured message.
+ * @param  measurement_record         A pointer to a destination buffer to store the HEM measurement record.
+ * @param  measurement_record_length  Indicate the size in bytes of the HEM measurement record.
+ * @param  mel_size                   On input, indicate the size in bytes of the destination buffer to store.
+ *                                    On output, indicate the size in bytes of the MEL.
+ * @param  measure_exten_log          A pointer to a destination buffer to store the MEL.
+ **/
+libspdm_return_t libspdm_get_measurement_extension_log(void *spdm_context,
```

Comment:
I think the API for this function should be
```
libspdm_return_t libspdm_get_measurement_extension_log(
void *spdm_context,
const uint32_t *session_id,
uint32_t offset,
uint32_t length,
size_t *mel_size,
void *mel);
```
and that libspdm should send as many `GET_MEASUREMENT_EXTENSION_LOG` messages as needed so that the returned `length` is greater than the desired `length`.

Reply:
In current design, the `libspdm_get_measurement_extension_log` will send many times in internal loop. 
I'm not sure why we need to add the offset and length parameters?

Reply:
Imagine the MEL is 100 MiB and the Requester retrieves the MEL. Some time later the MEL expands to 200 MiB. Why would the Requester request the initial 100 MiB again when it already has that information? When retrieving the second 100 MiB presumably it would start where it left off, and to do that it needs to specify the offset. Ie

1. Requester sets `offset = 0 && length = 500 MiB`.
     - Responder returns 100 MiB.
2. MEL grows to 200 MiB.
3. Requester sets `offset = 100 MiB && length = 500 MiB`.
     - Responder returns 100 MiB and Requester appends that to the previous 100 MiB MEL that was retrieved.

Reply:
After the discussion in https://github.com/DMTF/SPDM-WG/issues/3474 I think we should add a parameter, `bool continue_to_retrieve` or something, with the property
- `false`  (default) - libspdm caches the first `MEASUREMENT_EXTENSION_LOG.RemainderLength` and, if multiple `MEASUREMENT_EXTENSION_LOG` requests are needed then libspdm only retrieves up to that `RemainderLength`, even if the size of the MEL is growing through the multiple requests.
- `true` - libspdm will continue to fetch the MEL even if the MEL grows through multiple requests, up to the specified `length` value.

To start, we don't need to support the `true` path and can just make that parameter reserved. So 
```
libspdm_return_t libspdm_get_measurement_extension_log(
void *spdm_context,
const uint32_t *session_id,
bool reserved,
uint32_t offset,
uint32_t length,
size_t *mel_size,
void *mel);
```

Reply:
Discussed this at the SPDM Code Task Force meeting with @jyao1. He'd like the first API to be very simple and I agree. The above API can come later if needed and it would be `libspdm_get_measurement_extension_log_ex`. But all that is needed for the simple API is
```
void *spdm_context,
const uint32_t *session_id,
size_t *mel_size,
void *measure_exten_log
```
and libspdm retrieves the entire MEL from offset `0`.

Reply:
Thanks for your discussion. I have changed the API.